terraform-provider-constellation/internal/provider/cluster_resource.go

```diff
@@ -0,0 +1,308 @@
+	Attestation           types.Object `tfsdk:"attestation"`
+	OwnerID               types.String `tfsdk:"owner_id"`
+	ClusterID             types.String `tfsdk:"cluster_id"`
+	Kubeconfig            types.String `tfsdk:"kubeconfig"`
+	// NetworkConfig 			 types.Object `tfsdk:"network_config"` // TODO(elchead): do when clear what is needed
+}
+
+// Metadata returns the metadata of the resource.
+func (r *ClusterResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_cluster"
+}
+
+// Schema returns the schema of the resource.
+func (r *ClusterResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Resource for a Constellation cluster.",
+		Description:         "Resource for a Constellation cluster.",
+
+		Attributes: map[string]schema.Attribute{
+			"uid": schema.StringAttribute{
+				MarkdownDescription: "The UID of the cluster.",
+				Description:         "The UID of the cluster.",
+				Required:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Name used in the cluster's named resources / cluster name.",
+				Description:         "Name used in the cluster's named resources / cluster name.",
+				Optional:            true, // TODO(elchead): use "constell" as default
+			},
+			"image": schema.StringAttribute{
+				MarkdownDescription: "The Constellation OS image to use in the CSP specific reference format. Use the `constellation_image` data source to find the correct image for your CSP.",
+				Description:         "The Constellation OS image to use in the CSP specific reference format. Use the `constellation_image` data source to find the correct image for your CSP. When not set, the latest default version will be used.",
+				Optional:            true,
+			},
+			"kubernetes_version": schema.StringAttribute{
+				MarkdownDescription: "The Kubernetes version to use for the cluster.",
+				Description:         "The Kubernetes version to use for the cluster. When not set, the latest default version will be used.", // TODO(elchead): refer to supported versions; wait for https://github.com/edgelesssys/constellation/pull/2661
+				Optional:            true,
+			},
+			"constellation_microservices_version": schema.StringAttribute{
+				MarkdownDescription: "The Constellation microservices version to use for the cluster.",
+				Description:         "The Constellation microservices version to use for the cluster. When not set, the latest default version will be used.",
+				Optional:            true,
+			},
+			"debug": schema.BoolAttribute{
+				MarkdownDescription: "~> **Warning:** Do not enable Debug mode in production environments.\nEnable debug mode and allow the use of debug images.",
+				Description:         "DON'T USE IN PRODUCTION: Enable debug mode and allow the use of debug images.",
+				Optional:            true,
+			},
```

Comment:
Is there an option to mark attributes as hidden? If so we should do that for the `debug` mode.

Reply:
Looked at the code again.
I dont think we need this attribute for the Terraform provider at all.
The flag only changes how we deploy VMs (we use it to configure serial console access) and to print some warnings, so we should be ok with just removing it from the cluster resource

Reply:
deleted